charabia/src/tokenizer.rs

```diff
@@ -79,42 +79,40 @@ pub trait Tokenize<'o, A: AsRef<[u8]>> {
     /// assert_eq!(lemma, "quick");
     /// assert_eq!(kind, TokenKind::Word);
     /// ```
-    fn reconstruct(&self) -> ReconstructedTokenIter<A>;
+    fn reconstruct(&self) -> ReconstructedTokenIter;
 }
 
-impl Tokenize<'_, Vec<u8>> for &str {
-    fn tokenize(&self) -> NormalizedTokenIter<Vec<u8>> {
-        self.segment().classify().normalize(NormalizerOption::default())
+impl Tokenize<'_> for &str {
+    fn tokenize(&self) -> NormalizedTokenIter {
+        const NO: NormalizerOption =
+            NormalizerOption { create_char_map: false, stop_words: None, lossy: true };
+        self.segment().normalize(&NO)
     }
 
-    fn reconstruct(&self) -> ReconstructedTokenIter<Vec<u8>> {
+    fn reconstruct(&self) -> ReconstructedTokenIter {
         ReconstructedTokenIter { original: self, token_iter: self.tokenize() }
     }
 }
 
 /// Structure used to tokenize a text with custom configurations.
 ///
 /// See [`TokenizerBuilder`] to know how to build a [`Tokenizer`].
-pub struct Tokenizer<'al, 'sw, A> {
+pub struct Tokenizer<'al, 'no> {
     allow_list: Option<&'al HashMap<Script, Vec<Language>>>,
-    stop_words: Option<&'sw Set<A>>,
-    normalizer_option: NormalizerOption,
+    normalizer_option: &'no NormalizerOption<'no>,
 }
 
-impl<'al, 'sw, A: AsRef<[u8]>> Tokenizer<'al, 'sw, A> {
+impl<'al, 'no> Tokenizer<'al, 'no> {
     /// Creates an Iterator over [`Token`]s.
     ///
     /// The provided text is segmented creating tokens,
     /// then tokens are normalized and classified.
```

Comment:
Yes, I understand this. However, classifying tokens is no longer the responsibility of this function:

- previously, this function was explicitly calling `classify_with_stop_words` to classify the tokens
- now, this function classifies the tokens implicitly by calling `normalize`, because the classifier is part of the list of normalizers. If we were to remove the classifier from the list of normalizers the comment would no longer be true even though we're not modifying the implementation of this function.

I'm guessing the semantic issue I'm having is that `normalize` and the list of normalizers now can contain a classifier. Maybe we should have used a term encompassing both normalization and classification (but this is probably just `tokenize`, so redundant here in the comment).

Maybe a solution could be to specify:

```suggestion
    /// then tokens are normalized and classified depending on the list of normalizers and classifiers in [`normalizer::NORMALIZERS`].
```

Then the relationship between what happens and this list is made explicit. What do you think?